### PR TITLE
Implement TSMapEstimator benchmark

### PR DESCRIPTION
**Pull request by Jaleleddine**
It's a benchmark for tsmap_estimator. The results will be stored in the bench.yaml file.

### CODE REVIEW

**Comment by Jaleleddine** [comment]
done

File: benchmarks/tsmap_estimator.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        edisp=edisp,
+    )
+    return dataset
+
+
+def fit_estimator(dataset):
+    spatial_model = PointSpatialModel()
+    spectral_model = PowerLawSpectralModel(index=2)
+    model = SkyModel(spatial_model=spatial_model, spectral_model=spectral_model)
+    # estimator = TSMapEstimator(model)
```
Comment (registerrier) [inline benchmarks/tsmap_estimator.py:68]: remove commented line

File: benchmarks/tsmap_estimator.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+from gammapy.modeling.models import (
+    BackgroundModel,
+    SkyModel,
+    PowerLawSpectralModel,
+    PointSpatialModel,
+)
+from gammapy.irf import PSFMap, EnergyDependentTablePSF
+from gammapy.datasets import Datasets, MapDataset
+from gammapy.modeling.models import (
+    ExpCutoffPowerLawSpectralModel,
```
Comment (registerrier) [inline benchmarks/tsmap_estimator.py:22]: Import of models is done twice.  Remove this one.

File: benchmarks/tsmap_estimator.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import time
+import astropy.units as u
+import numpy as np
+import yaml
+from pathlib import Path
+
+from gammapy.maps import Map
+from gammapy.estimators import (
+    ASmoothMapEstimator,
```
Comment (registerrier) [inline benchmarks/tsmap_estimator.py:9]: Maybe you could also add a benchmarking for ASmoothMapEstimator

File: benchmarks/tsmap_estimator.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        "$GAMMAPY_DATA/fermi-3fhl-gc/fermi-3fhl-gc-background-cube.fits.gz"
+    )
+    background = BackgroundModel(background, datasets_names=["fermi-3fhl-gc"])
+
+    exposure = Map.read(
+        "$GAMMAPY_DATA/fermi-3fhl-gc/fermi-3fhl-gc-exposure-cube.fits.gz"
+    )
+    # unit is not properly stored on the file. We add it manually
+    exposure.unit = "cm2s"
+    mask_safe = counts.copy(data=np.ones_like(counts.data).astype("bool"))
```
Comment (adonath) [inline benchmarks/tsmap_estimator.py:40]: I think `mask_safe` is unused...just remove...

**Review by registerrier — COMMENTED** [review]
Thanks @Jaleleddine .

The PR also requires a small change in `make.py` to add this benchmark in the `AVAILABLE_BENCHMARKS` dictionary.

File: benchmarks/tsmap_estimator.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    background = Map.read(
+        "$GAMMAPY_DATA/fermi-3fhl-gc/fermi-3fhl-gc-background-cube.fits.gz"
+    )
+    background = BackgroundModel(background, datasets_names=["fermi-3fhl-gc"])
+
+    exposure = Map.read(
+        "$GAMMAPY_DATA/fermi-3fhl-gc/fermi-3fhl-gc-exposure-cube.fits.gz"
+    )
+    # unit is not properly stored on the file. We add it manually
+    exposure.unit = "cm2s"
```
Comment (adonath) [inline benchmarks/tsmap_estimator.py:39]: I think we fixed this and the unit should new read correctly from the file, but please check once again, before removing...

**Review by adonath — COMMENTED** [review]
Thanks @Jaleleddine, I left two minor comments. Once those are addressed we will merge and see if it integrates with the our CI setup...

**Comment by adonath** [comment]
@Jaleleddine Is this ready from your side?

**Pull request merged**